lib/navigation/control/navigator.cpp

```diff
@@ -9,56 +9,54 @@ Navigator::Navigator(core::ILogger &logger, const core::ITimeSource &time)
     : logger_(logger),
       time_(time),
       keyence_preprocessor_(logger),
+      optical_preprocessor_(logger),
       accelerometer_preprocessor_(logger, time),
-      accelerometer_trajectory_estimator_(time),
-      // crosschecker_(logger, time),
-      running_means_filter_(logger, time),
-      encoders_preprocessor_(logger),
-      previous_keyence_reading_({0, 0})
+      previous_accelerometer_data_(0.0),
+      previous_optical_reading_(0.0),
+      previous_keyence_reading_(0.0),
+      kalman_filter_(initial_state,       // If initial position not known exactly, modify
+                     initial_covariance,  // If initial position not known exactly, tune
```

Comment:
I think we can always say the initial position = 0, and run based on that. There will be some buffer

Also, this kind of thing should start `TODOLater`